* Implement divider for video "more dropdown menu"

PikachuEXE opened the pull request:
**Pull Request Type**
Please select what type of pull request this is:
- [x] Feature Implementation

**Related issue**
IRC chat:
```
Non-pressing question: if we were to ever have breakout sections on hover of the three dots video section dropdown, would the better organization be "Open >" & "Copy >" or "YouTube >" & "Invidious >"?
```

**Description**
- Update the dropdown opened by a button with title "More Options",  
  located on video card/box rendered in views like search result/history/trending,  
  so that the options are reordered and group together.  
  Dividers are used for grouping.  
- Menu item spacing reduced slightly.

**Screenshots (if appropriate)**
New screenshots (all dropdown updated by this PR, too lazy to take for all themes)
![image](https://user-images.githubusercontent.com/1018543/134648400-5aaaafa6-ad12-4abd-8b0b-591436e5c902.png)
![image](https://user-images.githubusercontent.com/1018543/134648415-025cfae3-a725-461f-885a-fa311dabbb72.png)
![image](https://user-images.githubusercontent.com/1018543/134648435-767d7eea-d3fc-4aed-8807-126441bd980e.png)


Old screenshots (before divider opacity reduced):
![image](https://user-images.githubusercontent.com/1018543/134623977-e441f182-4d36-44e7-b856-918b76d97fcb.png)
![image](https://user-images.githubusercontent.com/1018543/134623991-760c93fe-cf29-4322-bdeb-f28c1adc793c.png)
![image](https://user-images.githubusercontent.com/1018543/134623999-87765ca7-e78f-4bf8-b5da-a394cfac3159.png)
![image](https://user-images.githubusercontent.com/1018543/134624002-068ebc4e-9d52-4bb1-a2f3-ee6dbbc3bbd1.png)


**Testing (for code that is not small enough to be easily understandable)**
A: More Options
- Visit Trending
- Click "More Options" button
- Click on each item to if works as labeled

B: Download Video
- View a video (e.g. https://youtu.be/vyuJRJ6GQ5k)
- Click "Download Video" button
- Click on each item to if works as labeled (I only tested limited options)

C: Change Video Formats
- View a video (e.g. https://youtu.be/vyuJRJ6GQ5k)
- Click "Change Video Formats" button
- Click on each item to if works as labeled


**Desktop (please complete the following information):**
 - OS: MacOS
 - OS Version: 11.6
 - FreeTube version: 732b2d642df36f1417a3aae1a7867bd0b48fb54d

**Additional context**
Add any other context about the problem here.

PikachuEXE commented:
`dropdownValues` & `dropdownNames` should be removed eventually
Just not sure if you want me to update the other 2 places using `ft-icon-button` to use the new API (i.e. more code changes to review)

efb4f5ff-1298-471a-8973-3d47447115dc commented:
IIRC there is only one more dropdown like this


![FreeTube_wWk5jfbACw](https://user-images.githubusercontent.com/73130443/122607186-b9a9f980-d069-11eb-8bd7-6c8eb641fb74.png)

PikachuEXE commented:
2 more
- The download one you shown
- The video format one (DASH/Legacy/Audio) on the right of download button you pressed

PikachuEXE commented:
Updated all existing `ft-icon-button` users to use new API, updated PR description & screenshots about that
Not tested in all themes after divider style fix

efb4f5ff-1298-471a-8973-3d47447115dc commented:
found another dropdown https://youtube.com/playlist?list=PLF1CC814A59FF59DE

PikachuEXE commented:
@efb4f5ff-1298-471a-8973-3d47447115dc 
I assume you are referring to the button shown in the following screenshot
And [it's using `ft-list-dropdown`](https://github.com/FreeTubeApp/FreeTube/blob/development/src/renderer/components/playlist-info/playlist-info.vue#L42-L47) instead of `ft-icon-button` thus there is no need to update it

![image](https://user-images.githubusercontent.com/1018543/135785404-1f1236cc-df57-4921-a1b2-5f9d39a720d3.png)

peepopoggers commented:
I feel like this is an improvement, but it is a workaround for universal new share menu implimentation:
#1450

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (commented):
LGTM! Only thing i might change is in Dark mode it is hard to that an option is highlighted when u hover over it.

PikachuEXE commented:
Well I don't have good idea to fix that and that problem is not specific to this PR

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (dismissed):
Ur right, LGTM!

ChunkyProgrammer reviewed (dismissed):
LGTM

PikachuEXE commented:
code conflict 😨

PikachuEXE commented:
In progress (just restarted today)
Will take some time
Those code conflicts~~~
Well those actually resolved but now it's bugged lol

PikachuEXE commented:
Merge conflict fixed and whatever bug(s) I found fixed
But I have only tested the dropdown in screenshots (and I only test them briefly)
So please see if there are any more dropdowns that should be tested

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (approved):
LGTM

absidue reviewed (commented):
The [reduction of the vertical padding](https://github.com/FreeTubeApp/FreeTube/pull/1748/files#diff-1ac10679c04ebb500d84e838bb98cbf6c4f1c610a090bec30dd8ce37c9415cc0R86) around the menu options combined with the dividers make the longer options look quite cramped.

![bit_cramped](https://user-images.githubusercontent.com/48293849/170679792-fe3b6a8f-3f7e-4a63-924f-46458311c3a8.jpg)

I think 8px might be a good comprimise that still looks quite good without taking up too much space:
![8px](https://user-images.githubusercontent.com/48293849/170680841-c4f0edf4-84a7-44f6-9b42-448a5fc705fc.jpg)

Here is what it looks like with 10px:
![10px](https://user-images.githubusercontent.com/48293849/170680944-7137919c-ff98-4e80-900e-f087743cb128.jpg)

PikachuEXE commented:
@absidue 
8px now
Any functionality related bug found?

absidue commented:
I didn't find any functionality issues, thanks for updating the padding.

ChunkyProgrammer reviewed (approved):
LGTM
